### PR TITLE
Fold expressions

### PR DESCRIPTION
**Pull request by bernedom**
Added example for fold expressions to variadic template, would be glad for a review of th understandability of the example

### CODE REVIEW

File: variadic_templates.cc [hunk 1]
```diff
@@ -39,7 +62,20 @@ int main(int, char **) {
...
-  std::cout << sumSquares(1, 2, 3) << std::endl;
+  // returns "ABC"
+  std::cout << reorder_and_concat("CBA") << "\n";
+
+  // returns ABCJKL
+  std::cout << reorder_and_concat("CBA", "KLJ") << "\n";
+
+  // returns ABCJKLXYZ
+  std::cout << reorder_and_concat("CBA", std::string("KLJ"), "ZYX") << "\n"
+            << std::endl;
```
Comment (meshell) [inline variadic_templates.cc]: Remove the << std::endl (for consistency with the other outputs).

File: variadic_templates.cc [hunk 2]
```diff
@@ -1,34 +1,57 @@
...
+/*
+ * Supported operators are
+ * - boolean (&&, ||, <, >, ==, !=)
+ * - mathematical (+, -, *, /, %)
+ * - bitwise (&, |, ^, >>, <<)
+ * - memory (de-)referencing and obiect concatenation (.*, ->*)
+ * - assignement = and combinations such as /=, &=...
+ * - unpacking (,) [can be used to forward to normals functions]
+ */
+template <typename... Args> auto fold_using_add(Args &&... args) {
```
Comment (meshell) [inline variadic_templates.cc]: I would add multiple examples showing the different fold types. The example here is a _unary left fold_. An example for a _binary left fold_ could be
```
template <typename... Args> auto fold_using_add(Args &&... args) {
   return (0 + ... + args);
}
```
To show the difference between left and right fold you could use for example the following:
```
template <typename... Args> auto left_fold_minus(Args &&... args) {
    return (... - args);
}
// Example
std::cout << left_fold_minus(10, 3, 2) << "\n";       // (10 - 3) - 2 = 5

template <typename... Args> auto right_fold_minus(Args &&... args) {
    return (args - ...);
}
// Example
std::cout << right_fold_minus(10, 3, 2) << "\n";       // 10 - (3 - 2) = 9
```

**Review by meshell — COMMENTED** [review]
Add different examples for fold expressions (unary/binary) and (left/right). 
See comments.

File: variadic_templates.cc (hunk 1 above)
Comment (bernedom) [reply variadic_templates.cc]: Gone

File: variadic_templates.cc (hunk 2 above)
Comment (bernedom) [reply variadic_templates.cc]: Added example, I'm not too familiar with the fold expressions, @meshell  please confirm if I got the comments and the left/right correct.  
I updated the text in the booklet, but will add the example after your confirmation.

**Review by meshell — APPROVED** [review]

**Pull request merged**